src/Microsoft.OData.Edm/Csdl/Parsing/SchemaJsonParser.cs

```diff
@@ -0,0 +1,1564 @@
+                        {
+                            ReportUnexpectedElement(propertyValue, context);
+                        }
+
+                        break;
+                }
+            });
+
+            // If it's collection navigation property, let's construct the type name as Collection(entityType).
+            if (collection != null && collection.Value)
+            {
+                typeName = "Collection(" + typeName + ")";
+            }
+
+            // Annotations for $OnDelete are prefixed with $OnDelete. However, It's not supported in ODL yet.
+            if (csdlOnDelete != null && onDeleteAnnotations.Count > 0)
+            {
+                onDeleteAnnotations.ForEach(a => csdlOnDelete.AddAnnotation(a));
+            }
+
+            CsdlNavigationProperty csdlNavProperty = new CsdlNavigationProperty(name, typeName, nullable, parter,
+                containsTarget != null ? containsTarget.Value : false, // Absence of the member means false.
+                csdlOnDelete,
+                referentialConstraints != null ? referentialConstraints : Enumerable.Empty<CsdlReferentialConstraint>(),
+                context.Location());
+
+            navPropertyAnnotations.ForEach(a => csdlNavProperty.AddAnnotation(a));
+            return csdlNavProperty;
+        }
+
+        public static CsdlOnDelete ParseCsdlOnDelete(JsonElement element, JsonParserContext context)
+        {
+            string onDelete = element.ParseAsString(context);
+
+            EdmOnDeleteAction edmOnDelete = EdmOnDeleteAction.None;
+            switch (onDelete)
+            {
+                case "Cascade":
+                    edmOnDelete = EdmOnDeleteAction.Cascade;
+                    break;
+                default:
+                    // So far, ODL only supports "Cascade", for others "SetNull, or SetDefault.", we use "None" for them
+                    break;
+            }
+
+            return new CsdlOnDelete(edmOnDelete, context.Location());
+        }
+
+        private static IList<CsdlReferentialConstraint> ParseCsdlReferentialConstraint(JsonElement element, JsonParserContext context)
+        {
+            if (element.ValueKind != JsonValueKind.Object)
+            {
+                return null;
+            }
+
+            // The value of $ReferentialConstraint is an object with one member per referential constraint.
+            // The member name is the path to the dependent property, this path is relative to the structured type declaring the navigation property.
+            // The member value is a string containing the path to the principal property, this path is relative to the entity type that is the target of the navigation property.
+            // It also MAY contain annotations.These are prefixed with the path of the dependent property of the annotated referential constraint.
+            IList<CsdlReferentialConstraint> referentialConstraints = new List<CsdlReferentialConstraint>();

```

Comment:
>IList [](start = 12, length = 5)

"List"?

Reply:
List<T> implements IList<T> interface, so, what's the benefit to use "List"? 

---
In reply to: [443079003](https://github.com/OData/odata.net/pull/1775#discussion_r443079003) [](ancestors = 443079003)